addons/explosives/functions/fnc_interactEH.sqf

```diff
@@ -1,72 +1,73 @@
 /*
- * Author: PabstMirror
- * When interact_menu starts rendering (from "interact_keyDown" event)
- * Add defuse helpers to all nearby mines
+ * Author: PabstMirror, mharis001
+ * Dynamically adds "Defuse" actions to nearby mines when interact_menu is opened.
+ * Called by the "ace_interactMenuOpened" event.
  *
  * Arguments:
- * Interact Menu Type (0 - world, 1 - self) <NUMBER>
+ * Interact Menu Type (0 - World, 1 - Self) <NUMBER>
  *
  * Return Value:
  * None
  *
  * Example:
  * [0] call ace_explosives_fnc_interactEH
  *
- * Public: Yes
+ * Public: No
  */
 #include "script_component.hpp"
 
 params ["_interactionType"];
-TRACE_1("params",_interactionType);
+TRACE_1("Explosives interactEH",_interactionType);
 
-//Ignore self-interaction menu
-if (_interactionType != 0) exitWith {};
-//Ignore while mounted:
-if ((vehicle ACE_player) != ACE_player) exitWith {};
-//Ignore if we don't have defuse kit
-if (!("ACE_DefusalKit" in (items ACE_player))) exitWith {};
+// Ignore self-interaction menu or mounted vehicle interaction
+// For performance reasons only add PFH if player has a defusal kit
+// If player somehow gets a defusal kit during keyDown, they will just have to reopen menu
+if (
+    _interactionType != 0
+    || {vehicle ACE_player != ACE_player}
+    || {!("ACE_DefusalKit" in ([ACE_player, false, true, true, true, false] call CBA_fnc_uniqueUnitItems))}
+) exitWith {};
 
 [{
-    params ["_args", "_pfID"];
-    _args params ["_setPosition", "_addedDefuseHelpers", "_minesHelped"];
+    BEGIN_COUNTER(interactEH);
+    params ["_args", "_pfhID"];
+    _args params ["_setPosition", "_addedHelpers", "_minesHelped"];
 
     if (!EGVAR(interact_menu,keyDown)) then {
-        TRACE_1("Cleaning Defuse Helpers",(count _addedDefuseHelpers));
-        {deleteVehicle _x;} forEach _addedDefuseHelpers;
-        [_pfID] call CBA_fnc_removePerFrameHandler;
+        TRACE_1("Cleaning defuse helpers",count _addedHelpers);
+        {deleteVehicle _x} forEach _addedHelpers;
+        [_pfhID] call CBA_fnc_removePerFrameHandler;
     } else {
         // Prevent Rare Error when ending mission with interact key down:
-        if (isNull ace_player) exitWith {};
+        if (isNull ACE_player) exitWith {};
+        private _player = ACE_player;
```

Comment:
Accessing global variable is slower than local. So swap the order of the isNull and _player and reuse _player maybe.